health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/presentation/preparing/PreparingExerciseScreen.kt

```diff
@@ -0,0 +1,218 @@
+import androidx.compose.runtime.SideEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.health.services.client.data.LocationAvailability
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.wear.compose.material.Button
+import androidx.wear.compose.material.ButtonDefaults
+import androidx.wear.compose.material.Icon
+import androidx.wear.compose.material.MaterialTheme
+import androidx.wear.compose.material.Text
+import androidx.wear.compose.ui.tooling.preview.WearPreviewDevices
+import com.example.exercisesamplecompose.R
+import com.example.exercisesamplecompose.data.ServiceState
+import com.example.exercisesamplecompose.presentation.component.AcquiredCheck
+import com.example.exercisesamplecompose.presentation.component.NotAcquired
+import com.example.exercisesamplecompose.presentation.component.ProgressBar
+import com.example.exercisesamplecompose.presentation.dialogs.ExerciseInProgressAlert
+import com.example.exercisesamplecompose.presentation.theme.ThemePreview
+import com.example.exercisesamplecompose.service.ExerciseServiceState
+
+@Composable
+fun PreparingExerciseRoute(
+    onStart: () -> Unit,
+    onFinishActivity: () -> Unit,
+    onNoExerciseCapabilities: () -> Unit,
+) {
+    val viewModel = hiltViewModel<PreparingViewModel>()
+    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+
+    /** Request permissions prior to launching exercise.**/
+    val permissionLauncher = rememberLauncherForActivityResult(
+        ActivityResultContracts.RequestMultiplePermissions()
+    ) { result ->
+        if (result.all { it.value }) {
+            Log.d(TAG, "All required permissions granted")
+        }
+    }
+
+    SideEffect {
+        val preparingState = uiState
+        if (preparingState is PreparingScreenState.Preparing && !preparingState.hasExerciseCapabilities) {
+            onNoExerciseCapabilities()
+        }
+    }
+
+    if (uiState.serviceState is ServiceState.Connected) {
+        val requiredPermissions = uiState.requiredPermissions
+        LaunchedEffect(requiredPermissions) {
+            permissionLauncher.launch(requiredPermissions.toTypedArray())
+        }
+    }
+
+    Box(modifier = Modifier.fillMaxSize()) {
+        PreparingExerciseScreen(
+            onStart = {
+                viewModel.startExercise()
+                onStart()
+            },
+            uiState = uiState
+        )
+
+        if (uiState.isTrackingInAnotherApp) {
+            val dismissed by remember { mutableStateOf(false) }
+            ExerciseInProgressAlert(onNegative = onFinishActivity, onPositive = {}, !dismissed)
```

Comment:
I added `dismissed = true` in the brackets and changed dismissed to a var. Was not seeing the dialog get dismissed otherwise!

Reply:
Otherwise, everything works great!

Reply:
Damn, sorry about that lazy code.  Thanks for the thorough review.